assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -13287,6 +13313,14 @@ protected int sizeResources(SessionState state) {
                         return (tagIds.containsAll(selectedTags));
                     }).collect(Collectors.toList());
                 }
+                
+                //Search assignments by title
+                String searchTerm = (String) state.getAttribute(SEARCH_ASSIGNMENTS);
+                if (searchTerm != null && !searchTerm.isEmpty()) {
+                    returnResources = ((List<Assignment>)returnResources).stream()
+                        .filter(a -> a.getTitle().toLowerCase().contains(searchTerm.toLowerCase()))
```

Comment:
```suggestion
                        .filter(a -> a.getTitle().equalsIgnoreCase(searchTerm))
```

Reply:
Hi @ern , I’ve just pushed a new commit reverting your suggestion regarding the search filter. Based on Andrea's comments, the change makes the search box less useful, as it only matches exact titles. The original implementation, which allows for partial matches, seems to be more aligned with the intended functionality.